src/System.IO.Pipelines/tests/BackpressureTests.cs

```diff
@@ -29,39 +29,6 @@ public void Dispose()
 
         private readonly Pipe _pipe;
 
-        [Fact]
-        public void AdvanceThrowsIfFlushActiveAndNotConsumedPastThreshold()
```

Comment:
Turn this into tests for the new behavior. We don't have any.

Reply:
We kinda do though, the ones that test backpressure today pass the same thing for consumed an examined.

Reply:
No. We need a test that consumes nothing but advances everything past the backpressure limit and make sure writer is unblocked.

Reply:
We only have tests that either consume and examine at the same time or not hit backpressure.

Reply:
Sure, that sounds like a single test 😄 .

Reply:
Wrote a more complicated test and found a bug